### ISSUE TITLE
texture used for filling Line2D get's strangely distorted when put on tile

### ISSUE THREAD

**Issue by ajreckof** [godotengine/godot#70337]
### Godot version

4.0.beta8 

### System information

MacOS Monterey 12.4 puce M1 Pro

### Issue description

when trying to do a dotted line i used a Line2D and used a small circle with blank spaces on both sides but when i tried to make it repeat the texture was distorted. This only happens when multiples line are drawn with the same line2D it seems to comes from the angle 

### Steps to reproduce

open the minimal reproduction project
 move around the lines 
enjoy the weirdness of it all

### Minimal reproduction project

[Archive 2.zip](https://github.com/godotengine/godot/files/10266233/Archive.2.zip)

**Comment by TheSofox** [comment]
Attaching screenshots for context: 
![image](https://github.com/godotengine/godot/assets/3089525/501d1576-42e8-4c62-9b07-19700aa9250b)
![image](https://github.com/godotengine/godot/assets/3089525/cb9b2554-5917-4ee3-8f8f-377a92ac50c0)

I'm guessing this is another affine texture warping issue, not unlike in #83306 (thought that one was caused by width_curve, I believe it's the same fundamental problem).

**Comment by MewPurPur** [comment]
Yeah, like I explained, it's because when you subtract the joints from the equation, the segments are trapezoid.

**Comment by feiyuhuahuo** [comment]
This bug still exists in Godot4.3, hope it can be fix soon.

**Comment by mmewtwosaysbye** [comment]
Still getting it in 4.5
